lineapy/graph_reader/program_slice.py

```diff
@@ -49,11 +63,94 @@ def get_slice_graph(
     for sink in sinks:
         ancestors.update(graph.get_ancestors(sink))
 
+    ancestors = _include_dependencies_for_indirectly_included_nodes_in_slice(
+        graph, ancestors
+    )
+
     new_nodes = [graph.ids[node] for node in ancestors]
     subgraph = graph.get_subgraph(new_nodes)
     return subgraph
 
 
+def _include_dependencies_for_indirectly_included_nodes_in_slice(
```

Comment:
None of the code in program slicing seems to be tested 😟 so I have no template to work with.
I feel that itself is worthy of an epic seeing the amount of untested code. 
Thoughts?

Reply:
I think we should definitely look at our code coverage. I believe we actually print this information in pre-commit but haven't consulted this information much. 

Let's make sure not to pile on the testing debt by adding tests for this piece of logic. We can create synthetic graphs and test for inclusion. It would just be like any other unit test. Shouldn't be too gnarly since the function is stateless and we don't need to mock anything.